### PR TITLE
DOC: Migrate and update docs for building ITK Python wheels

### PR DESCRIPTION
**Pull request by tbirdso**
Migrate documentation for building ITK Python distributions and build archives with ITKPythonPackage.

Original source: https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/master/docs/Build_ITK_Python_packages.rst

Includes updates for ITK v5.3.0 and removed obsolete `sdist` instructions.

## PR Checklist
- [x] Updated API documentation (or API not changed)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Documentation/docs/contributing/python_packaging.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+Build ITK Python packages
+======================================
+
+We can generate Python wrappings and package them as Python wheels in order to conveniently distribute and use the ITK C++ library in Python scripts. In most cases community members can use the [general ITK Python distributions](https://pypi.org/project/itk/) on the Python Package Index (PyPI) for their projects.
+
+```py
+> pip install itk
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:7]: If we can remove `>`, when rendered there will be will a copy butten, and we can copy directly from the rendered docs.

File: Documentation/docs/contributing/python_packaging.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+$ git clone https://github.com/InsightSoftwareConsortium/ITKPythonPackage.git
+[...]
+
+$ pushd ITKPythonPackage
+$ export MANYLINUX_VERSION=2014
+$ ./scripts/dockcross-manylinux-build-wheels.sh cp38
+[...]
+
+$ ls -1 dist/
+itk-5.3.0.dev20231108-cp38-cp38m-manylinux2014_x86_64.whl
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:47]: Can we remove the `$`?

File: Documentation/docs/contributing/python_packaging.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+First, install the Python.org macOS Python distributions. This step requires sudo:
+
+```sh
+./scripts/macpython-install-python.sh
+```
+
+Then, run [macpython-build-wheels.sh](https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/master/scripts/macpython-build-wheels.sh) to build the wheels. A Python version may be targetd by passing a trailing argument to the script. See ITKPythonPackage for environment variables used by `macpython-build-wheels.sh`.
+
+```sh
+$ git clone https://github.com/InsightSoftwareConsortium/ITKPythonPackage.git
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:61]: $

File: Documentation/docs/contributing/python_packaging.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+[...]
+
+$ ./scripts/macpython-build-wheels.sh cp38
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:64]: $

File: Documentation/docs/contributing/python_packaging.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+[...]
+```
+
+## Uploading to PyPI
+
+InsightSoftwareConsortiums may upload ITK package wheels to PyPI using `twine`.
+
+First, install dependencies:
+```py
+> python -m pip install twine
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:107]: $

File: Documentation/docs/contributing/python_packaging.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+Then, upload packages to the [PyPI testing server](https://test.pypi.org):
+```py
+> python -m twine upload -r pypitest dist/*
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:112]: $

File: Documentation/docs/contributing/python_packaging.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+After verifying distributions appear correctly on the testing server, upload the packages to the production PyPI server:
+
+```py
+> python -m twine upload dist/*
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:118]: $

File: Documentation/docs/contributing/python_packaging.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+The updated ITK Python wheels are now available for general installation:
+```py
+python -m pip install itk
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:123]: $

File: Documentation/docs/contributing/python_packaging.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+## `ITKPythonBuilds` Archives
+
+In addition to PyPI releases, the InsightSoftwareConsortium releases ITK build archives at [ITKPythonBuilds/releases](https://github.com/InsightSoftwareConsortium/ITKPythonBuilds/releases). Those archives contain ITK sources and binaries required for building ITK external modules with ITK Python wrappings.
+
+Scripts are available in ITKPythonPackage to generate build archives on each platform after automated builds are generated above:
+
+### Linux
+
+```sh
+$ ./scripts/dockcross-manylinux-build-tarball.sh
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:135]: $

File: Documentation/docs/contributing/python_packaging.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+### macOS
+
+```sh
+$ ./scripts/macpython-build-tarball.sh
```
Comment (thewtex) [inline Documentation/docs/contributing/python_packaging.md:141]: $

**Review by thewtex — CHANGES_REQUESTED** [review]
Looks great!

If we can remove the prefixes from the shell commands, they will be copy-able from the UI.

File: Documentation/docs/contributing/python_packaging.md (hunk 2 above)
Comment (tbirdso) [reply Documentation/docs/contributing/python_packaging.md:47]: It is useful to understand the build command output here. Do you have an alternate suggestion on how to convey input/output?

Comment (thewtex) [reply Documentation/docs/contributing/python_packaging.md:47]: Yes, we could keep these.

**Comment by tbirdso** [comment]
Received offline approval to merge from @thewtex 👍

**Pull request merged**